virttest/utils_misc.py

```diff
@@ -634,16 +634,20 @@ def get_full_pci_id(pci_id):
         return None
 
 
-def get_pci_id_using_filter(pci_filter, session=None):
+def get_pci_id_using_filter(pci_filter, session=None, domain=False):
     """
     Get PCI ID from pci filter in host or in guest.
 
     :param pci_filter: PCI filter regex of a device (adapter name)
     :param session: vm session object, if none use host pci info
+    :param domain: boolean, if True include the domain in the pci id output
 
     :return: list of pci ids with adapter name regex
     """
-    cmd = "lspci | grep -F '%s' | awk '{print $1}'" % pci_filter
+    if domain:
```

Comment:
Well, I think can use `get_full_pci_id`, but as I know, getting a pci_id without domain is not useful. Also, lspci will only hide the domain id when it starts with `0000:`.

Reply:
What do you think @YongxueHong

Reply:
For some command, it works with/without domain id, such as 'lspci -k -s', so I think we can support both.
and get_pci_id_using_filter is using by other cases.

Reply:
Only when the domain ID is `0000` we can ignore it, otherwise we cannot get the correct one.
```bash
$ lspci -ks 00:04.0
0004:00:04.0 PCI bridge: Ampere Computing, LLC Altra PCI Express Root Port a3 (rev 04)
	Kernel driver in use: pcieport
000c:00:04.0 PCI bridge: Ampere Computing, LLC Altra PCI Express Root Port a3 (rev 04)
	Kernel driver in use: pcieport
000d:00:04.0 PCI bridge: Ampere Computing, LLC Altra PCI Express Root Port a3 (rev 04)
	Kernel driver in use: pcieport
```

Reply:
Hi, @quanwenli 
From my understanding of your patching code, your requirement is that you would like to add an argument `domain` to get a full PCI ID: `$domain: $bus: $device: $function`, please correct me if I misunderstood.
So as @PaulYuuu mentioned https://github.com/avocado-framework/avocado-vt/pull/3901/files#r1599264989 , we could call the existing method `def get_full_pci_id(pci_id):` to cover your requirement. 
We don't need to implement the same functionality on both functions.
Thanks.

Reply:
Ok, I have removed the domain, and add the support for session in method of get_full_pci_id. please help review.